### PR TITLE
Fix: An array can have only integers or strings as keys

### PR DESCRIPTION
**Pull request by voku**
We should allow all kind of string types here. We currently do not have an array-key interface or something like that, so I added the new type into the check.

### CODE REVIEW

File: src/Types/ClassString.php [hunk 1]
```diff
@@ -14,14 +14,13 @@
...
 use phpDocumentor\Reflection\Fqsen;
-use phpDocumentor\Reflection\Type;
 
 /**
  * Value Object representing the type 'string'.
  *
  * @psalm-immutable
  */
-final class ClassString implements Type
+final class ClassString extends String_
```
Comment (jaapio) [inline src/Types/ClassString.php]: I think we forgot this one when introducing the PseudoType interface, Could you please add that as well here.

**Review by jaapio — CHANGES_REQUESTED** [review]
Thanks for this improvement, it would be nice if you can update the PR with my requested change. 

After that, this should be ready to merge.

File: src/Types/ClassString.php (hunk 1 above)
Comment (voku) [reply src/Types/ClassString.php]: done

File: src/Types/ArrayKey.php [hunk 2]
```diff
@@ -13,20 +13,28 @@
...
+final class ArrayKey extends AggregatedType implements PseudoType
 {
     public function __construct()
     {
         parent::__construct([new String_(), new Integer()], '|');
     }
 
+    public function underlyingType(): Type
+    {
+        return new String_();
```
Comment (jaapio) [inline src/Types/ArrayKey.php]: This seems incorrect to me, this type is not a string, it should return a `Compound` with the types `String_` and `Integer`. ArrayKey is an alternative notation for `String|Int`, not just `String`

Comment (voku) [reply src/Types/ArrayKey.php]: fixed

**Review by jaapio — APPROVED** [review]
Thanks for yet another nice contribution!

**Pull request merged**